### PR TITLE
210 errors and trees

### PR DESCRIPTION
**Pull request by paulp**
Review by @jsuereth.

### CODE REVIEW

**Comment by xeno-by** [comment]
Thanks for `makeAnonymousNew`

File: src/compiler/scala/tools/nsc/ast/parser/TreeBuilder.scala [hunk 1]
```diff
@@ -199,6 +199,15 @@ abstract class TreeBuilder {
     }
   }
 
+  /** Creates a tree representing new Object { stats }.
+   *  To make sure an anonymous subclass of Object is created,
+   *  if there are no stats, a () is added.
+   */
+  def makeAnonymousNew(stats: List[Tree]): Tree = {
```
Comment (jsuereth) [inline src/compiler/scala/tools/nsc/ast/parser/TreeBuilder.scala:206]: Thanks much for this!

File: test/files/neg/names-defaults-neg.check [hunk 2]
```diff
@@ -76,7 +76,7 @@ and  method f in object t8 of type (a: Int, b: Object)String
 match argument types (a: Int,b: String) and expected result type Any
   println(t8.f(a = 0, b = "1")) // ambigous reference
              ^
-names-defaults-neg.scala:69: error: wrong number of arguments for <none>: (x: Int, y: String)A1
+names-defaults-neg.scala:69: error: wrong number of arguments for pattern A1(x: Int,y: String)
```
Comment (jsuereth) [inline test/files/neg/names-defaults-neg.check:79]: This is much nicer.

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**